Ali | Amir / Storybook for components (#2009)
binary-com/deriv-app: packages/components/stories/contract-card/README.md

# ContractCard Component

> Shows detailed information of a contract. Two types of contracts are available to use: `multiplier` contract and `simple` contracts. A contract can be either `completed` or `ongoing`. 

Reviewer: ```suggestion
Shows detailed information of a contract. Two types of contracts are available to use: `multiplier` contract and `simple` contract. A contract can be either `completed` or `ongoing`. 
```

> Renders a `ContractCard` component showing the information of a contract. Two types of contracts defined in this component are multiplier contract and not-multiplier contract. A contract may be completed or still ongoing. 

Reviewer 2: ```suggestion
Renders a `ContractCard` component which shows detailed information of a contract. Two types of contracts are available to use: `multiplier` and `simple` contracts. A contract might have two states: `completed` or `ongoing`. 
```

---

## ContractCard Header
Renders a `Header` for the contract card

```jsx
import ContractCard from 'Components/contract-card';

const DummyComponent = props => (
    <ContractCard contract_info={data.contract_info}>
        <ContractCard.Header
            contract_info={data.contract_info}
            getCardLabels={getCardLabels}
            getContractTypeDisplay={getContractTypeDisplay}
            server_time={null}
        />
    </ContractCard>
);
```

#### Props

| Name                    | Type         | Default     | Description                                                 |
| ----------------------- | ------------ | ----------- | ----------------------------------------------------------- |
| contract_info           | {object}     | null        | The information of the contract                             |
| getCardLabels           | {function}   | null        | Function returns the labels to show in the card             |
| getContractTypeDisplay  | {function}   | null        | Function returns the display name of the contract type      |
| has\_progress\_slider   | {boolean}    | null        | Defines visibility of the sidebar. For ongoing contracts    |

> | is_mobile               | {boolean}    | null        | Showes whether the device is mobile or not                  |
> | is\_sell\_requested     | {boolean}    | null        | Showes whether the user requested for sell or not           |
> | is_sold                 | {boolean}    | null        | Showes whether the contract is sold or not                  |

Reviewer 2: ```suggestion
| is_mobile               | {boolean}    | null        | Shows whether the device is mobile or not                  |
| is\_sell\_requested     | {boolean}    | null        | Shows whether the user requested for sell or not           |
| is_sold                 | {boolean}    | null        | Shows whether the contract is sold or not                  |
```

const DummyComponent = props => (
    <ContractCard contract_info={data.contract_info}>
        <ContractCard.Body
            contract_info={data.contract_info}
            contract_update={data.contract_update}
            currency={data.currency}
            getCardLabels={getCardLabels}
            is_mobile={false}
            is_multiplier={data.is_multiplier}
            status={data.status}
            server_time={toMoment(1603881751)}
        />
    </ContractCard>
);
```

#### Props

| Name             | Type         | Default     | Description                                                 |
| ---------------- | ------------ | ----------- | ----------------------------------------------------------- |
| contract_info    | {object}     | null        | The information of the contract                             |
| getCardLabels    | {function}   | null        | Function returns the labels to show in the card             |
| currency         | {string}     | null        | The contract currency                                       |
| is_multiplier    | {boolean}    | null        | Defines the type of contract                                |

> | is_mobile        | {boolean}    | null        | Showes whether the device is mobile or not                  |
> | is_sold          | {boolean}    | null        | Showes whether the contract is sold or not                  |

Reviewer 2: ```suggestion
| is_mobile        | {boolean}    | null        | Shows whether the device is mobile or not                  |
| is_sold          | {boolean}    | null        | Shows whether the contract is sold or not                  |
```

            contract_info={data.contract_info}
            getCardLabels={getCardLabels}
            server_time={data.server_time}
            status={data.status}
        />
    </ContractCard>
);
```

#### Props

| Name                               | Type         | Default     | Description                                                         |
| ---------------------------------- | ------------ | ----------- | ------------------------------------------------------------------- |
| contract_info                      | {object}     | null        | The information of the contract                                     |
| getCardLabels                      | {function}   | null        | Function returns the labels to show in the card                     |
| getContractById                    | {function}   | null        | Function returns the contract information                           |
| is_multiplier                      | {boolean}    | null        | Defines the type of contract                                        |
| is_positions                       | {boolean}    | null        | Defines whether the contract is position or not                     |

> | is\_sell\_requested                | {boolean}    | null        | Showes whether the user requested for sell or not                   |

Reviewer 2: Same here
